operator-common/src/main/java/io/strimzi/operator/common/operator/resource/BuildConfigOperator.java

```diff
@@ -38,4 +40,43 @@ public BuildConfigOperator(Vertx vertx, OpenShiftClient client) {
         // Cascading needs to be set to false to make sure the Builds are not deleted during reconciliation
         return super.internalPatch(namespace, name, current, desired, false);
     }
+
+    /**
+     * Asynchronously deletes the resource in the given {@code namespace} with the given {@code name},
+     * returning a Future which completes once the {@code delete} returns.
+     *
+     * This is n override for BuildConfigs because the {@code selfClosingWatch} used by {@code AbstractResourceoperator} does not work for them.
```

Comment:
Do we know why it doesn't work?

Reply:
I did not dig into the details on OCP side TBH. But it seems that the BuildConfig API behaves differently and just kicks the watch when the resource is deleted instead of sending the DELETED event.